### PR TITLE
HTTP and test environment dependencies up to date

### PR DESCRIPTION
**Pull request by JorgeCastilloPrz**
### Rationale

I was just about to start using your client for a sample project and found that it was so out of date and using many old library versions. Would be ❤️ If you can have this merged and deployed to maven central in a new version at some point.

### Work done
* Upgraded the following dependencies:
  * Junit 4.11 -> 4.12
  * MockWebServer 2.5.0 -> 2.7.5 .
  * apache commons-io 2.4 -> 2.5
  * OkHttp 2.5.0 -> 3.6.0
  * Retrofit 2.0.0-beta2 -> 2.2.0
  * Retrofit gson converter 2.0.0-beta2 -> 2.2.0
  * OkHttp logging interceptor 2.6.0 -> 3.6.0

* Updated sources (main and tests ones) to match new library requirements.

### Hints
I needed to add the following lines to the library `build.gradle` file in order to copy test resources to the proper folder:
```groovy
sourceSets {
  test {
    output.resourcesDir = "build/classes/test/resources"
  }
}
```
For some reason, test resources where not being properly copied by default and some tests where failing with null pointer exceptions. Android studio still is not copying the resources by itself when I run tests using the IDE (not sure about why tbh). But yeah, tests are passing when I run them using the gradle console: `./gradlew :MarvelApiClient:test`. 

I hope that's enough.

### CODE REVIEW

**Review by pedrovgs — APPROVED** [review]
Thank you so much @JorgeCastilloPrz !! You have my 👍 but I'd like to wait for @flipper83's review before to merge this PR and release a new version.

**Comment by JorgeCastilloPrz** [comment]
Don't worry, lets wait for him! :)

**Comment by JorgeCastilloPrz** [comment]
Ping! 

Any updates about this?

**Comment by pedrovgs** [comment]
I'm merging this PR and releasing a new version.

**Comment by pedrovgs** [comment]
@JorgeCastilloPrz version 1.0.0 released. It will be available soon. Thanks for your contribution!!! 😃

**Comment by JorgeCastilloPrz** [comment]
coool. Thanks guys!

**Pull request merged**
